.github/actions/amazon-linux-build-action/Dockerfile

```diff
@@ -5,13 +5,13 @@ FROM amazonlinux:latest
 RUN yum install -y tar gzip gcc
 
 # install Go
-ENV GOLANG_VERSION 1.16.6
-ENV GOLANG_VERSION_SHA256 be333ef18b3016e9d7cb7b1ff1fdb0cac800ca0be4cf2290fe613b3d069dfe0d
+ENV GOLANG_VERSION 1.18.3
+ENV GOLANG_VERSION_SHA256 956f8507b302ab0bb747613695cdae10af99bbd39a90cae522b7c0302cc27245
 
-RUN curl -o golang.tar.gz https://dl.google.com/go/go$GOLANG_VERSION.linux-amd64.tar.gz \
-	&& echo "$GOLANG_VERSION_SHA256 golang.tar.gz" | sha256sum --strict --check \
-	&& tar -C /usr/local -xzf golang.tar.gz \
-	&& rm golang.tar.gz
+RUN curl -O https://dl.google.com/go/go$GOLANG_VERSION.linux-amd64.tar.gz \
```

Comment:
As the commit comment says, I had to do this, because renaming messes up sha256 sum, so it is different from the one published on [the official website](https://go.dev/dl/). I could of course download the package, rename it and calculate the new hash, but I don't think it is the way to go.